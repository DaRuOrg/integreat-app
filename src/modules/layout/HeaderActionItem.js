/**
 * HeaderActionItem is the data class which needs to be supplied to HeaderActionBar.
 */
class HeaderActionItem {
  constructor ({iconSrc, href, dropDownElement}) {
    this._iconSrc = iconSrc
    this._href = href
    this._dropDownElement = dropDownElement
  }

  get iconSrc () {
    return this._iconSrc
  }

  get href () {
    return this._href
  }

  get dropDownElement () {
    return this._dropDownElement
  }
}

export default HeaderActionItem
