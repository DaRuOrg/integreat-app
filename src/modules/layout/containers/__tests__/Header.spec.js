import React from 'react'
import { shallow, mount } from 'enzyme'
import configureMockStore from 'redux-mock-store'

import HeaderActionItem from '../../HeaderActionItem'
import HeaderNavigationItem from '../../HeaderNavigationItem'
import ConnectedHeader, { Header } from '../Header'
import { Provider } from 'react-redux'

const mockStore = configureMockStore()

describe('Header', () => {
  test('should match snapshot with smallViewport', () => {
    const component = shallow(
      <Header smallViewport
              actionItems={[new HeaderActionItem({href: '/action1'})]}
              navigationItems={[new HeaderNavigationItem({href: '/nav1'})]} />
    )
    expect(component).toMatchSnapshot()
  })

  test('should match snapshot with largeViewport', () => {
    const component = shallow(
      <Header smallViewport={false}
              actionItems={[new HeaderActionItem({href: '/action1'})]}
              navigationItems={[new HeaderNavigationItem({href: '/nav1'})]} />
    )
    expect(component).toMatchSnapshot()
  })

  test('should connect to the store', () => {
    const smallStore = mockStore({viewport: {is: { small: true }}})
    const smallComponent = mount(<Provider store={smallStore}>
        <ConnectedHeader />
    </Provider>)

    expect(smallComponent.find(Header).prop('smallViewport')).toBe(true)

    const largeStore = mockStore({viewport: {is: { small: false }}})
    const largeComponent = mount(<Provider store={largeStore}>
      <ConnectedHeader />
    </Provider>)
    expect(largeComponent.find(Header).prop('smallViewport')).toBe(false)
  })
})
