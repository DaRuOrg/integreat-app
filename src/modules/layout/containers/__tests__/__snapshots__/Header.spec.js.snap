// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match snapshot with largeViewport 1`] = `
<Headroom
  pinStart={0}
  scrollHeight={85}
>
  <header
    className="header"
  >
    <div
      className="logoWide"
    >
      <img
        src="test-file-stub"
      />
    </div>
    <HeaderActionBar
      className="actionBar"
      items={
        Array [
          HeaderActionItem {
            "_dropDownNode": undefined,
            "_href": "/action1",
            "_iconSrc": undefined,
          },
        ]
      }
    />
    <HeaderMenuBar
      className="navigationBar"
      items={
        Array [
          HeaderNavigationItem {
            "_active": undefined,
            "_href": "/nav1",
            "_text": undefined,
          },
        ]
      }
    />
  </header>
</Headroom>
`;

exports[`Header should match snapshot with smallViewport 1`] = `
<Headroom
  pinStart={0}
  scrollHeight={55}
>
  <header
    className="header"
  >
    <div
      className="logoWide"
    >
      <img
        src="test-file-stub"
      />
    </div>
    <HeaderActionBar
      className="actionBar"
      items={
        Array [
          HeaderActionItem {
            "_dropDownNode": undefined,
            "_href": "/action1",
            "_iconSrc": undefined,
          },
        ]
      }
    />
    <HeaderMenuBar
      className="navigationBar"
      items={
        Array [
          HeaderNavigationItem {
            "_active": undefined,
            "_href": "/nav1",
            "_text": undefined,
          },
        ]
      }
    />
  </header>
</Headroom>
`;
